### PR TITLE
PiP Project changes

### PR DESCRIPTION
**Pull request by komalm**

### CODE REVIEW

File: site/models/items.php [hunk 1]
```diff
@@ -306,4 +311,35 @@ private function isValidDate($date)
...
+	 * Function that finds Joomla group id from group name
+	 * Move this function to pip helper
+	 *
+	 * @param   String  $groupName  groupName
+	 *
+	 * @return  group ID
+	 *
+	 * @since  1.0.0
+	 */
+	public function getUsersGroupId($groupName)
```
Comment (coolbung) [inline site/models/items.php]: This function is not needed. Joomla has a JField for user groups. Check how it's done there.

File: site/models/items.php [hunk 2]
```diff
@@ -306,4 +311,35 @@ private function isValidDate($date)
...
+		$db     = JFactory::getDBO();
+		$query  = $db->getQuery(true);
+		$query->select('*');
+		$query->from('#__usergroups');
+		$db->setQuery($query);
+		$groups = $db->loadRowList();
+
+		foreach ($groups as $group)
+		{
+			if ($group[4] == $groupName)
```
Comment (coolbung) [inline site/models/items.php]: Why is this hardcoded to 4 ?

File: site/views/itemform/tmpl/details.php [hunk 3]
```diff
@@ -21,140 +21,133 @@
...
+											<?php echo $field->label; ?>
+										</div>
 
-											<div class="col-sm-6 control-label">
-											<?php
+										<div class="col-sm-6 control-label">
+										<?php
+											$fileType = explode(".", $field->value);
+											$arr_image_type = array('jpeg', 'JPEG', 'png', 'PNG', 'jpg','JPG');
+											if (in_array(end($fileType), $arr_image_type))
```
Comment (coolbung) [inline site/views/itemform/tmpl/details.php]: Suggest you pick up the allowed types from a config.

File: site/views/itemform/tmpl/details.php [hunk 4]
```diff
@@ -21,140 +21,133 @@
...
-												$arr_image_type = array('jpeg', 'JPEG', 'png', 'PNG', 'jpg','JPG');
-												if (in_array(end($fileType), $arr_image_type))
+										<div class="col-sm-6 control-label">
+										<?php
+											foreach ($field->value as $val)
+											{
+												foreach ($val as $lab => $valu)
 												{
-													echo '<div><img height="80" width="100" src="' . JUri::root(true) . $field->value . '"></div>';
+													$db    = JFactory::getDbo();
```
Comment (coolbung) [inline site/views/itemform/tmpl/details.php]: We stopped adding queries in layout files 👎

File: site/models/items.php (hunk 2 above)
Comment (parthlawate) [inline site/models/items.php]: temp solution till we implement the correct approach using permissions. Creating privacy issues on the live site now so added as a temp fix

Comment (coolbung) [inline site/models/items.php]: Ok. Can't the temp solution be to hardcode this in the view instead. The models can stay clean.

Comment (parthlawate) [inline site/models/items.php]: Yes can do.. @komalm assuming this was done for the permissions ?

Comment (komalm) [inline site/models/items.php]: yes sir.

File: site/models/items.php (hunk 1 above)
Comment (komalm) [inline site/models/items.php]: Removed this function and used user group table object.

File: site/models/items.php (hunk 2 above)
Comment (komalm) [inline site/models/items.php]: Removed this function.

**Pull request merged**